tests/unit/utils/test_gitignore.py

```
@@ -0,0 +1,66 @@
+    assert "generated.txt" not in results
+    assert ".venv/ignored.py" not in results
+    assert "ignored-dir/skip.py" not in results
+
+
+def test_list_cwd_falls_back_when_gitignore_is_unreadable(
+    tmp_path: Path,
+    monkeypatch,
+) -> None:
+    original_read_text = Path.read_text
+
+    def _raise_permission_error(self: Path, *args, **kwargs) -> str:
```

Ruff (0.15.7)

[warning] Missing type annotation for `*args`

(ANN002)